legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabProjectApi.java

```diff
@@ -203,43 +211,131 @@ public Project createProject(String name, String description, String groupId, St
-            gitLabApi.getProtectedBranchesApi().protectBranch(gitLabProject.getId(), defaultBranch, AccessLevel.NONE, AccessLevel.MAINTAINER);
+        // try to ensure current user is a maintainer (or higher), but proceed even if we fail
+        AccessLevel minUserAccessLevel = AccessLevel.MAINTAINER;
+        if (!tryEnsureMinAccessLevel(gitLabApi, gitLabProject, minUserAccessLevel))
+        {
+            LOGGER.warn("Created project {} but could not set access level for {} to {} - trying to proceed anyway", project.getProjectId(), getCurrentUser(), minUserAccessLevel);
+        }
 
-            // build project structure
-            String projectId = GitLabProjectId.getProjectIdString(getGitLabConfiguration().getProjectIdPrefix(), gitLabProject);
+        // build project structure
+        try
+        {
             int projectStructureVersion = getDefaultProjectStructureVersion();
             ProjectConfigurationUpdater configUpdater = ProjectConfigurationUpdater.newUpdater()
-                    .withProjectId(projectId)
+                    .withProjectId(project.getProjectId())
                     .withGroupId(groupId)
                     .withArtifactId(artifactId)
                     .withProjectStructureVersion(projectStructureVersion);
             if (this.projectStructurePlatformExtensions != null)
             {
                 configUpdater.setProjectStructureExtensionVersion(this.projectStructureExtensionProvider.getLatestVersionForProjectStructureVersion(projectStructureVersion));
             }
-            ProjectStructure.newUpdateBuilder(getProjectFileAccessProvider(), projectId, configUpdater)
+            ProjectStructure.newUpdateBuilder(getProjectFileAccessProvider(), project.getProjectId(), configUpdater)
                     .withMessage("Build project structure")
                     .withProjectStructureExtensionProvider(this.projectStructureExtensionProvider)
                     .withProjectStructurePlatformExtensions(this.projectStructurePlatformExtensions)
                     .build();
+        }
+        catch (Exception e)
+        {
+            throw buildException(e,
+                    () -> "User " + getCurrentUser() + " is not allowed to set up project structure for newly created project " + project.getProjectId(),
+                    () -> "Error setting up project structure for newly created project " + project.getProjectId(),
+                    () -> "Error setting up project structure for newly created project " + project.getProjectId());
+        }
 
-            return fromGitLabProject(gitLabProject);
+        // protect default branch
+        String defaultBranch = getDefaultBranch(gitLabProject);
+        try
+        {
+            ProtectedBranchesApi protectedBranchesApi = gitLabApi.getProtectedBranchesApi();
+            withRetries(() -> protectedBranchesApi.protectBranch(gitLabProject.getId(), defaultBranch, AccessLevel.NONE, AccessLevel.MAINTAINER));
```

Comment:
Was the expectation before this change that gitlab would protect the default branch upon creation?

Reply:
No, the code to protect the branch was there previously, just in a different place.